packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Thenable} from 'react-reconciler/src/ReactFiberScheduler';
+
+import warningWithoutStack from 'shared/warningWithoutStack';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+// Keep in sync with ReactDOMUnstableNativeDependencies.js
+// ReactDOM.js, and ReactTestUtils.js:
+const [
+  /* eslint-disable no-unused-vars */
+  getInstanceFromNode,
+  getNodeFromInstance,
+  getFiberCurrentPropsFromNode,
+  injectEventPluginsByName,
+  eventNameDispatchConfigs,
+  accumulateTwoPhaseDispatches,
+  accumulateDirectDispatches,
+  enqueueStateRestore,
+  restoreStateIfNeeded,
+  dispatchEvent,
+  runEventsInBatch,
+  /* eslint-enable no-unused-vars */
+  flushPassiveEffects,
+] = ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.Events;
+
+const RendererHelpers = {
```

Comment:
Why this extra wrapper?

Reply:
Ok so I have a constraint that the 'rest' of the file should be copy paste-able across implementations. In the Noop renderer versions, because of a cyclical import, I cannot 'read' the values `flushPassiveEffects` and `batchedUpdates` off the renderer at module loading time (ie - `flushPassiveEffects` is not available on module start, so I have to do `ReactNoop.flushPassiveEffects()` instead of `flushPassiveEffects()` later). To unify the 2 constraints, I made this object, which shares its shape (for these 2 functions) with the renderer.  (So in the Noop renderer, I just do `import RendererHelpers from 'ReactNoop'`)Thought it was an elegant solution. Happy to hear if you have an alternate idea!

Reply:
The simplest solution to this is to just not have a separate file for ReactNoopAct. Just put it in the same file.

Reply:
figured it was too clever by half ha. will do.